docs/code-conventions/sql.md

```diff
@@ -0,0 +1,178 @@
+- Columns that hold ID/key values generated by another system should always unambiguously identify that system.
+    - You may feel that it's obvious to everybody that `external_id` or `source_id` refers to the company's "SplineReticulator Directory" product.
+      But even if that were true _at the time of writing_, that does not mean that the same is true for _future you_ or somebody else in charge of deciphering the code in 5 years.
+    - Also, following this rule makes it easier to connect additional systems (each with their own column), or migrate from one to the other.
+- As columns are nullable by default, omit the `null` keyword in column definitions.
+
+## Datatypes
+
+### Textual data: `text` vs. `varchar`
+
+- In PostgreSQL and other RDBMS that support it, use the `text` datatype for any textual column, irrespective of the length of expected values.
+  Unlike `varchar(x)`, the `text` datatype does not need (or support) a predefined length, which makes code more maintainable.
+
+    !!! note "Implementing length restrictions"
+        If you _do_ need to enforce a length restriction on the database level, use a check constraint instead.
+        This way, changing the length restriction later on will not cause the RDBMS to physically rewrite the entire table.
+
+### Using `decimal` and `numeric`
+
+- In PostgreSQL, our default choice for storing numbers of any kind is to use `numeric` without specifying a precision or scale.
+  Such columns can store several thousands of digits both before and after the decimal point.[^1]
+  Not having to define precision or scale in advance makes our SQL code more maintainable.
+
+    !!! warning "Non-standard behavior"
+        ANSI SQL specifies that leaving out the precision or scale is equivalent to `(0)`, i.e. integer values.
+        Still, the PostgreSQL behavior is so useful that we accept this tradeoff in portability.
+
+- Only when there is a good reason to have the database coerce or round input values to a certain precision or scale, use `decimal(p,s)` with appropriate values.
+  One such reason, obviously, is that the respective project does not use PostgreSQL.
+
+- Although `numeric` and `decimal` are synonyms, we use each exclusively for the purpose given above in order to make our intentions more obvious.
+  Therefore, never use `numeric(p,s)` (with scale and precision) or `decimal` (without scale and precision).
+
+[^1]: See the paragraph on "unconstrained numeric" columns in the [numeric datatypes](https://www.postgresql.org/docs/current/datatype-numeric.html#DATATYPE-NUMERIC-DECIMAL) chapter of the PostgreSQL docs.
+
+### Boolean values
+
+- In PostgreSQL and other RDBMS that support it or an equivalent datatype, we use `boolean` for all boolean-valued columns. 
+    - Note that by default, such columns allow `null` in addition to `true` and `false`.
+      Specify the column with a `not null` clause to prevent this.
```

Comment:
I feel, this "note" is not strong enough.
```suggestion
    - Always specify `boolean` columns as `not null` unless `null` is forced upon us by the source of the data (i.e. an upstream system).
    Allowing `null` makes it harder to use the column later on, as it ends up having three possible states instead of two, leaving it up to debate whether `null` means `false` or `true`.
```

Reply:
Good point, I'll make that change.

Reply:
Done. I also added a rule that when null cannot be avoided, the meaning of null values should be documented.